### PR TITLE
tests: Fix waitall with empty sendbuf, add minor

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

File: xdsl/dialects/mpi.py [hunk 1]
```diff
@@ -212,10 +212,10 @@ class Allreduce(MPIBaseOp):
 
     name = "mpi.allreduce"
 
+    send_buffer: Annotated[OptOperand, Attribute]
     recv_buffer: Annotated[Operand, Attribute]
     count: Annotated[Operand, i32]
     datatype: Annotated[Operand, DataType]
-    send_buffer: Annotated[OptOperand, Attribute]
```
Comment (webmiche) [inline xdsl/dialects/mpi.py:218]: I must admit, I am surprised that reordering this changes something and I would consider that an xDSL-bug. I think we should investigate this more deeply to understand why the reodering changes something. Could you reduce the testcase and file an issue? Or maybe @math-fehr has a better idea what's happening?

Comment (superlopuh) [reply xdsl/dialects/mpi.py:218]: Yeah this seems bad.

File: xdsl/dialects/mpi.py [hunk 2]
```diff
@@ -230,9 +230,9 @@ def get(
             SSAValue | Operation | Sequence[SSAValue | Operation]
         ] = []
         if send_buffer is None:
-            operands_to_add = [recv_buffer, count, datatype, []]
+            operands_to_add = [[], recv_buffer, count, datatype]
```
Comment (superlopuh) [inline xdsl/dialects/mpi.py:233]: Why not pass None?

File: xdsl/dialects/mpi.py (hunk 1 above)
Comment (webmiche) [reply xdsl/dialects/mpi.py:218]: From this [discussion](https://xdsl.zulipchat.com/#narrow/stream/368604-Framework/topic/IRDL.20.7C.20OptOperand.20vs.20Variadics) it seems that this is an op specific problem and not an xDSL related one.

File: xdsl/dialects/mpi.py (hunk 2 above)
Comment (superlopuh) [reply xdsl/dialects/mpi.py:233]: Yeah, although I have a slight preference towards None, since it communicates that it's optional and not variadic

**Comment by tobiasgrosser** [comment]
This looks nice. I guess the use of None/not-none discussion of @superlopuh  and @georgebisbas  seems to be the only remaining issue. Otherwise, this seems good to go. Thank you @georgebisbas

File: xdsl/dialects/mpi.py (hunk 2 above)
Comment (georgebisbas) [reply xdsl/dialects/mpi.py:233]: I see.

Comment (superlopuh) [reply xdsl/dialects/mpi.py:233]: This is a nit, I don't feel strongly about this :)

**Review by superlopuh — APPROVED** [review]

**Pull request merged**